pkg/vgmanager/vgmanager_controller.go

```diff
@@ -394,12 +399,86 @@ func (r *VGReconciler) addDevicesToVG(vgName string, devices []internal.BlockDev
 }
 
 // filterMatchingDevices returns unmatched and matched blockdevices
-// TODO: Implement this
-func filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, []internal.BlockDevice, error) {
-	// currently just match all devices
+func (r *VGReconciler) filterMatchingDevices(blockDevices []internal.BlockDevice, volumeGroup *lvmv1alpha1.LVMVolumeGroup) ([]internal.BlockDevice, []internal.BlockDevice, error) {
```

Comment:
Unit test should be added for this function.

Reply:
comment still valid.

Reply:
I will add them later once we agree with the actual code changes to minimize the changes in unit tests.